### ISSUE TITLE
Memory leak on demo asio_httpd

### ISSUE THREAD

**Issue by mingodad** [waruqi/tbox#8]
Hello !
I just cloned tbox and build it, then when testing ./demo asio_httpd with httpress/ab and watching a task manager I can see how memory consumption only grows, keep running the command bellow and watch process memory usage:
```
ab -n 100000 -c 2000 -k http://127.0.0.1:8080/tbox/config.h

or

httpress -t 2 -n 100000 -c 2000 -k http://127.0.0.1:8080/tbox/config.h
``` 

I tested with valgrind and it reports no memory leaks (same number of allocations/deallocations) when terminating the program normally (key press), so I think it's memory fragmentation or any pool attached to a connection that is not been released (but because maybe they are attached to a hierarchy of pools that are released at the end valgrind can not find anything). 

The system where it's running is a ubuntu 14.04 64bits.

Cheers !

**Comment by mingodad** [comment]
Running this command the program starts with 20MB and when served 5,000,000 requests it's using around 180MB and continue growing but I stopped the program.
```
echo 1 > 1.txt
./demo asio_httpd

on other console:
httpress -t 2 -n 10000000 -c 2000 -k http://127.0.0.1:8080/1.tx
``` 
Cheers !

**Comment by mingodad** [comment]
Also trying requesting a non existing file is faster to see the memory growing:
```
httpress -t 2 -n 10000000 -c 2000 -k http://127.0.0.1:8080/xxx.tx
```

**Comment by mingodad** [comment]
I also tried with other parameters at demo.c:
if (!tb_init(tb_null, tb_native_allocator())) return 0;
if (!tb_init(tb_null, tb_static_allocator((tb_byte_t*)malloc(300 * 1024 * 1024), 300 * 1024 * 1024))) return 0;

But the same memory consumption pattern happen.

Cheers !

**Comment by foobar** [comment]
can you be more specific on the memory usage?  Was it virtual space or RSS?

**Comment by mingodad** [comment]
Hello !
It's RSS .
Couldn't you reproduce it with the same/similar load test ?

Cheers !

**Comment by waruqi** [comment]
I will try to fix this problem.

Thanks!

**Comment by waruqi** [comment]
Hi!

I have fixed this memory issue with commits: 79f496a and a6d6b1f

Thanks!

**Comment by mingodad** [comment]
Not at all !
Thank you for this great work !
